pkg/client/options.go

```diff
@@ -521,6 +524,18 @@ func (c Continue) ApplyToList(opts *ListOptions) {
 	opts.Continue = string(c)
 }
 
+// UnsafeDisableCacheDeepCopy indicates not to deep copy objects during list objects in cache reader.
+type UnsafeDisableCacheDeepCopy struct {
+	// if IgnoreGVK is true, cache reader will always list without deep copy.
+	// Otherwise, it will only avoid deep copy for objects with correct groupVersionKind.
+	IgnoreGVK bool
```

Comment:
I don't understand this option, the cache is gvk indexed anyways?

Reply:
The `IgnoreGVK` is used [here](https://github.com/kubernetes-sigs/controller-runtime/pull/1274/files#diff-7e7ff90d22397cbc25370857a4c26baa1062866c5c9581e440d1e37603e45011R169). For I see the objects will be `SetGroupVersionKind` in [previous logic](https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/cache/internal/cache_reader.go#L154), it seems some objects have no GVK when they are listed from informer.
So if this `IgnoreGVK` is true, the objects will still be deepcopy if they have no GVK when listed from informer, because they need to be `SetGroupVersionKind`.
I don’t know if I explain it clearly...

Reply:
This is a bug in the deserialization: https://github.com/kubernetes/kubernetes/issues/80609

We work around this in some places by manually setting GVK.

IMHO this option should not exist, as it is not controllable by users if TypeMeta is filled. We should just ignore TypeMeta intead.